terminus-ui/chip/src/chip-collection.component.ts

```diff
@@ -0,0 +1,569 @@
+   * @returns True if the index is valid for our collection of chips.
+   */
+  private isValidIndex(index: number): boolean {
+    return index >= 0 && index < this.chips.length;
+  }
+
+
+  /** When blurred, mark the field as touched when focus moved outside the chip collection. */
+  public blur() {
+    if (!this.hasFocusedChip()) {
+      this.keyManager.setActiveItem(-1);
+    }
+  }
+
+  private resetChips() {
+    this.dropSubscriptions();
+    this.listenToChipsFocus();
+    this.listenToChipsSelection();
+    this.listenToChipsRemoved();
+  }
+
+  private dropSubscriptions() {
+    if (this.chipSelectionSubscription) {
+      this.chipSelectionSubscription.unsubscribe();
+      this.chipSelectionSubscription = null;
+    }
+
+    if (this.chipFocusSubscription) {
+      this.chipFocusSubscription.unsubscribe();
+      this.chipFocusSubscription = null;
+    }
+
+    if (this.chipBlurSubscription) {
+      this.chipBlurSubscription.unsubscribe();
+      this.chipBlurSubscription = null;
+    }
+
+    if (this.chipRemoveSubscription) {
+      this.chipRemoveSubscription.unsubscribe();
+      this.chipRemoveSubscription = null;
+    }
+  }
+
+  /** Listens to user-generated selection events on each chip. */
+  private listenToChipsSelection(): void {
+    this.chipSelectionSubscription = this.chipSelectionChanges.subscribe(event => {
+      event.source.selected
+        ? this.selectionModel.select(event.source)
+        : this.selectionModel.deselect(event.source);
+
+      // For single selection chip collection, make sure the deselected value is unselected.
+      if (!this.allowMultipleSelections) {
+        this.chips.forEach(chip => {
+          if (!this.selectionModel.isSelected(chip) && chip.selected) {
+            chip.deselect();
+          }
+        });
+      }
+    });
+  }
+
+  /** Listens to user-generated selection events on each chip. */
+  private listenToChipsFocus(): void {
+    this.chipFocusSubscription = this.chipFocusChanges.subscribe(event => {
+      const chipIndex: number = this.chips.toArray().indexOf(event.chip);
+
+      if (this.isValidIndex(chipIndex)) {
+        this.keyManager.updateActiveItem(chipIndex);
+      }
+    });
+
+    this.chipBlurSubscription = this.chipBlurChanges.subscribe(() => {
+      this.blur();
+    });
+  }
+
+  private listenToChipsRemoved(): void {
+    this.chipRemoveSubscription = this.chipRemoveChanges.subscribe(event => {
+      const chip = event.chip;
+      const chipIndex = this.chips.toArray().indexOf(event.chip);
+
+      // In case the chip that will be removed is currently focused, we temporarily store
+      // the index in order to be able to determine an appropriate sibling chip that will
+      // receive focus.
+      if (this.isValidIndex(chipIndex) && chip.hasFocus) {
+        this.lastDestroyedChipIndex = chipIndex;
+      }
+    });
+  }
+
+  /** Checks whether any of the chips is focused. */
+  private hasFocusedChip() {
```

Comment:
(btw, this is only a suggestion, not a blocker for the PR)

IMO it makes the intention more clear. When I call a function, there is a 50/50 chance I'm triggering functionality. But when I interact with a getter I'm just requesting a piece of data. (of course, people can abuse getters to do more than return data) I've noticed many of these types of functions are even  named like a getter eg `getActiveChip` etc. 

Again, might be personal preference, but 
`myChip = foo.activeChip` 
seems slightly more concise than 
`myChip = foo.getActiveChip()`

Also I find it clear because a function may or may not return anything, but a getter must return something.